### DISCUSSION TITLE
PGroonga 2.4.5 リリース!

### DISCUSSION THREAD

**Discussion by komainu8** [pgroonga/pgroonga#297] [Releases]
## PGroonga（ぴーじーるんが） 2.4.5リリース

PostgreSQLで高速日本語全文検索をできるようにする[PGroonga](https://pgroonga.github.io/ja/)の2.4.5をリリースしました！

### 修正

* [[`pgroonga_text_regexp_ops_v2 operator class`](https://pgroonga.github.io/ja/reference/#text-regexp-ops-v2)] normalizersなどのオプションを指定していると、 `pgroonga_text_regexp_ops_v2` を指定していても `TokenRegexp` が使われない問題を修正しました。[#293][Reported by askdkc]

* [[`Crash safe`](https://pgroonga.github.io/ja/reference/crash-safe.html)] GroongaのWALによる復旧が失敗したときに、PGroongaのインデックスの復旧に失敗する問題を修正しました。

### [[`Crash safe`](https://pgroonga.github.io/ja/reference/crash-safe.html)] GroongaのWALによる復旧が失敗したときに、PGroongaのインデックスの復旧に失敗する問題を修正しました。

PGroongaのクラッシュセーフは、PGroongaのインデックスの変更点をGroongaのWALでに随時書き出しています。 PostgreSQLのクラッシュなどで正常にシャットダウンされなかった時に、書き出していたGroongaのWALを使って復旧を試みますが、GroongaのWALでも復旧に失敗した場合は、既存のGroongaのデータベースを削除して、新しくGroongaのデータベースを作り、 `REINDEX` を実行しPostgreSQLのデータからPGroongaのインデックスを再生成します。

今回の問題は、上記のようにGroongaのWALを使った復旧に失敗したあとの `REINDEX` が正しく実行されていませんでした。 結果として、GroongaのWALを使った復旧に失敗すると、PGroongaのインデックスは破損したままになります。

## まとめ

新規ユーザーの方は、[PGroongaについて](https://groonga.org/ja/blog/2017/10/10/pgroonga-2.0.2.html#pgroonga)も参照してください。

PostgreSQLで高速に日本語全文検索をしたいという方はPGroongaを使ってガンガン検索してください！

**Comment by askdkc** [comment]
HomeBrewのGitHubに[PR出てる](https://github.com/Homebrew/homebrew-core/pull/124899/)のに中々落ちてこないな〜と思っていたら、

https://github.com/Homebrew/homebrew-core/pull/124899/commits/0c5c7ed5579ec9ac37396a3cb7b5b8f66caa0035

```rb
 def postgresql
     Formula["postgresql@15"]
 end
```

が原因でテストエラーになってる。。。

その後、修正されたようです

https://github.com/Homebrew/homebrew-core/pull/124899/commits/1f26014e6d57bfb47c8b5c4e3797fbf8dc2e919d

**Comment by askdkc** [comment]
@kou さん

HomebrewのPRのやり取りを見ると

https://pgroonga.github.io/news/#version-2-4-1

でPostgreSQL 15がサポートされたと書いてあるのに`Formula/pgroonga.rb`で

```rb
  def postgresql
    Formula["postgresql@15"]
  end
```

にするとhomebrew側のテストでコケちゃいますね

macOSやUbuntuパッケージ版はPostgreSQL 14でサポートと明記しないと混乱する？

**Reply by kou** [reply]
チラ見しましたが、PGroongaの問題じゃなくてHomebrewのformulaの問題な気がします。

（コメントしておきました。）

**Reply by askdkc** [reply]
@kou さん

ありがたや〜

ちなみにmacOS版も本当はPostgreSQL 15でもそのまま動く感じです？
（今はPostgreSQL 15にしてもPGroongaが認識されないみたいで動かない）

**Reply by kou** [reply]
試していないですけど、動いて欲しいです。LinuxとWindowsで動いているのでmacOSでだけ動かないってことはないと思います。

認識されないというのはHomebrewで入れたやつの話ですか？それとも自分でビルドしたやつの話ですか？

**Reply by askdkc** [reply]
@kou さん

brewで入れた方です

PostgreSQL 14だと普通に動くのですが、PostgreSQL 15にするとパスに存在しないだかなんだかみたいなエラーが出ます
（今ちょっと手元に開発用Macが無くて詳細確認出来ずですが）

**Reply by kou** [reply]
であれば、それは↑のPRで直るやつだと思います。

**Comment by askdkc** [comment]
@kou さん

PGroonga 2.4.6を準備中とは思うのですが、macOS版のbrewの2.4.5がまだエラーで止まってるみたいです😵

https://github.com/Homebrew/homebrew-core/actions/runs/4421707884/jobs/7752879125?pr=124899#step:7:136

`Errno::EPERM: Operation not permitted @ rb_file_s_rename - (/opt/homebrew/opt/postgresql@15/lib/libecpg.6.15.dylib, /opt/homebrew/Cellar/pgroonga/2.4.5/lib/libecpg.6.15.dylib) `

**Reply by kou** [reply]
https://github.com/Homebrew/homebrew-core/pull/124899#issuecomment-1482273943
パッチを置いておきました。
